Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -236,22 +236,26 @@ internal extension URL {
     }
 }
 
-public extension HTTPClient {
-    final class Task<Response> {
+extension HTTPClient {
+    public final class Task<Response> {
         public let eventLoop: EventLoop
-        let future: EventLoopFuture<Response>
+        public let promise: EventLoopPromise<Response>
 
         private var channel: Channel?
         private var cancelled: Bool
         private let lock: Lock
 
-        init(eventLoop: EventLoop, future: EventLoopFuture<Response>) {
+        public init(eventLoop: EventLoop) {
             self.eventLoop = eventLoop
-            self.future = future
+            self.promise = eventLoop.makePromise()
             self.cancelled = false
             self.lock = Lock()
         }
 
+        public var futureResult: EventLoopFuture<Response> {
```

Comment:
I think if `promise` is public, we probably don't need this or the succeed, wait, fail, etc methods.

Reply:
actually, as Ian pointed out, it's better to have promise internal, so users cannot accidentally succeed it. Other methods are just convenience helpers